### ISSUE TITLE
Suggest a replacement in `package_hooks_linter`?

### ISSUE THREAD

**Issue by etiennebacher** [r-lib/lintr#2662]
Some `lintr` messages show the potential replacement for a linter, e.g. in `any_duplicated_linter`:

``` r
library(lintr)
lint(
  text = "any(duplicated(x), na.rm = TRUE)",
  linters = any_duplicated_linter()
)
#> <text>:1:1: warning: [any_duplicated_linter] anyDuplicated(x, ...) > 0 is better than any(duplicated(x), ...).
#> any(duplicated(x), na.rm = TRUE)
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

However, some messages are not very helpful because they don’t recommend any replacement. For example, `package_hooks_linter` says `Don't use cat() in .onAttach().`:

``` r
lint(
  text = ".onLoad <- function(lib, pkg) cat('hi')",
  linters = package_hooks_linter()
)
#> <text>:1:31: warning: [package_hooks_linter] Don't use cat() in .onLoad().
#> .onLoad <- function(lib, pkg) cat('hi')
#>                               ^~~
```
Is it possible to suggest the replacement for this linter? My reflex was to use `message()` instead but this is also flagged. Same with `packageStartupMessage()` (in `.onLoad()` only, not in `.onAttach()`) so I don’t know what should be used here.

**Comment by MichaelChirico** [comment]
messages should be produced by .onAttach, not .onLoad

**Comment by etiennebacher** [comment]
Then I think messages from `.onLoad` should say something like:

> Don't use cat() in .onLoad(). Use `packageStartupMessage()` in `.onAttach()` instead.

and messages from `.onAttach()` could say:

> Don't use cat() in .onAttach(). Use `packageStartupMessage()` instead.

**Comment by AshesITR** [comment]
I suggest using a more active lint message for `.onAttach`:

> Use `packageStartupMessage()` instead of `cat()` in .onAttach().
